broker/lib/jobs/BnRStatusPollerJob.js

```diff
@@ -122,11 +132,116 @@ class BnRStatusPollerJob extends BaseJob {
+    const plan = catalog.getPlan(instanceInfo.plan_id);
+    return Promise.try(() => {
+        if (operationName === CONST.OPERATION_TYPE.BACKUP) {
+          return lockManager.lock(instance_guid, {
+              lockedResourceDetails: {
+                resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.BACKUP,
+                resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+                resourceId: backup_guid,
+                operation: CONST.OPERATION_TYPE.BACKUP
+              }
+            })
+            .catch(err => {
+              if (err instanceof DeploymentAlreadyLocked) {
+                logger.info(`Proceeding as lock is already acquired for the resource: ${instance_guid}`);
+              } else {
+                throw err;
+              }
+            })
+            .then(() => {
+              return BackupManager.createManager(plan)
+                .then(backupManager => backupManager.getOperationState(CONST.OPERATION_TYPE.BACKUP, instanceInfo));
+            });
+        }
+      })
+      .then(operationStatusResponse => {
+        operationStatusResponse.jobCancelled = false;
+        operationStatusResponse.operationTimedOut = false;
+        operationStatusResponse.operationFinished = false;
+        if (utils.isServiceFabrikOperationFinished(operationStatusResponse.state)) {
+          operationStatusResponse.operationFinished = true;
+          return operationStatusResponse;
+        } else {
+          logger.info(`Instance ${instance_guid} ${operationName} for backup guid ${backup_guid} still in-progress - `, operationStatusResponse);
+          const currentTime = new Date();
+          const backupTriggeredDuration = (currentTime - new Date(instanceInfo.started_at)) / 1000;
+          return Promise
+            .try(() => eventmesh
+              .apiServerClient
+              .patchOperationResponse({
+                operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
+                operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+                operationId: instanceInfo.backup_guid,
+                value: operationStatusResponse
+              }))
+            .then(() => bosh.director.getDirectorConfig(instanceInfo.deployment))
+            .then(directorConfig => {
+              const lockDeploymentMaxDuration = directorConfig.lock_deployment_max_duration;
+              if (backupTriggeredDuration > lockDeploymentMaxDuration) {
+                //Operation timed out
+                if (!instanceInfo.abortStartTime) {
+                  //Operation not aborted. Aborting operation and with abort start time
+                  // re-registering statupoller job
+                  let abortStartTime = new Date().toISOString();
+                  instanceInfo.abortStartTime = abortStartTime;
+                  return BackupManager.createManager(plan)
+                    .then(directorManager => directorManager.abortLastBackup(instanceInfo, true))
```

Comment:
It would be a good idea to hit the abort endpoint instead of calling this method directly to decouple job with Backupmanager.